spec/classes/vault_spec.rb

```diff
@@ -6,10 +8,8 @@
   end
 
   on_supported_os.each do |os, facts|
-    context "on #{os} " do
-      let :facts do
-        facts.merge(service_provider: 'init', processorcount: 3)
-      end
+    context "on #{os}" do
+      let(:facts) { override_facts(facts, service_provider: 'systemd', processors: { count: 3 }) }
```

Comment:
Rather than overriding the service provider, I'd consider relying on https://github.com/voxpupuli/modulesync_config/commit/607bf0021177cd16e50d902cdba7e82a8b381213 which means it should exist (and have a correct value) as long as stdlib is in the dependencies.